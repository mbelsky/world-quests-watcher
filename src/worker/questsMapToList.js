export function questsMapToList(map) {
  return Object.values(map).reduce((quests, value) => {
    quests.push(...value);

    return quests;
  }, []);
}
