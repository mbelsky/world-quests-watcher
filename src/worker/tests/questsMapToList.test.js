import { questsMapToList } from "../questsMapToList.js";

test("map convertation to list", () => {
  const map = {
    eu: [
      {
        endsAt: "11 hours 54 minutes",
        href: "/quest=52986",
        id: "52986",
        region: "eu",
      },
      {
        endsAt: "11 hours 54 minutes",
        href: "/quest=52988",
        id: "52988",
        region: "eu",
      },
    ],
    na: [
      {
        endsAt: "11 hours 54 minutes",
        href: "/quest=52986",
        id: "52986",
        region: "na",
      },
      {
        endsAt: "11 hours 54 minutes",
        href: "/quest=52989",
        id: "52989",
        region: "na",
      },
    ],
  };

  expect(questsMapToList(map)).toMatchSnapshot();
});
