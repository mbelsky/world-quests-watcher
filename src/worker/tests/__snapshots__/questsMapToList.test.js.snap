// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`map convertation to list 1`] = `
Array [
  Object {
    "endsAt": "11 hours 54 minutes",
    "href": "/quest=52986",
    "id": "52986",
    "region": "eu",
  },
  Object {
    "endsAt": "11 hours 54 minutes",
    "href": "/quest=52988",
    "id": "52988",
    "region": "eu",
  },
  Object {
    "endsAt": "11 hours 54 minutes",
    "href": "/quest=52986",
    "id": "52986",
    "region": "na",
  },
  Object {
    "endsAt": "11 hours 54 minutes",
    "href": "/quest=52989",
    "id": "52989",
    "region": "na",
  },
]
`;
